### ISSUE TITLE
Systemd file not working

### ISSUE THREAD

**Issue by sascha-wi** [atuinsh/atuin#2280]
### What did you expect to happen?

Easy to set up atuin self-hosted server

### What happened?

When starting the server it gives gibberish error:
`(atuin)[38065]: atuin-server.service: Failed at step NAMESPACE spawning atuin: No such file or directory
(atuin)[545]: atuin-server.service: Failed to set up mount namespacing: /run/systemd/unit-root/etc/atuin: No such file or directory`

```bash
cat /etc/systemd/system/atuin-server.service
[Unit]
Description=Start the Atuin server syncing service
After=network-online.target
Wants=network-online.target systemd-networkd-wait-online.service

[Service]
ExecStart=atuin server start
Restart=on-failure
User=atuin
Group=atuin

Environment=ATUIN_CONFIG_DIR=/etc/atuin
ReadWritePaths=/etc/atuin

# Hardening options
CapabilityBoundingSet=
AmbientCapabilities=
NoNewPrivileges=true
ProtectHome=true
ProtectSystem=strict
ProtectKernelTunables=true
ProtectKernelModules=true
ProtectControlGroups=true
PrivateTmp=true
PrivateDevices=true
LockPersonality=true

[Install]
WantedBy=multi-user.target
```

### Atuin doctor output

```yaml
Atuin Doctor
Checking for diagnostics


Please include the output below with any bug reports or issues

{
  "atuin": {
    "version": "18.3.0",
    "sync": null,
    "sqlite_version": "3.44.0"
  },
  "shell": {
    "name": "bash",
    "default": "bash",
    "plugins": [
      "atuin",
      "bash-preexec"
    ],
    "preexec": "bash-preexec"
  },
  "system": {
    "os": "Debian GNU/Linux",
    "arch": "x86_64",
    "version": "12",
    "disks": [
      {
        "name": "/dev/vda1",
        "filesystem": "ext4"
      }
    ]
  }
}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by sascha-wi** [comment]
I assume the error lies somewhere here: 
Environment=ATUIN_CONFIG_DIR=/etc/atuin
ReadWritePaths=/etc/atuin

What do I need to set these variables to when I install atuin through cargo? Can you please update your self-hosting documentation?

**Comment by ghostheory** [comment]
@sascha-wi I got this working, but it was a bit of a mess. 

I installed atuin using the install script with my normal user (on a ubuntu headless server in my homelab) which dropped the atuin binary into the `~/.atuin/bin` directory, which is not at all helpful for getting the server going. 

So, I was getting the same error you are seeing above. To get this thing to run I had to:
  - create the atuin user in `/etc/sysusers.d/atuin-server.conf` for systemd to have a system user in place to use.
  - create the `/etc/atuin` directory
  - chown/chmod the `/etc/atuin` directory so it was owned by the `atuin` system user I just created (otherwise systemd chokes on bad permissions)
  - `cp` the atuin binary to a path found in `$PATH` so systemd can run it. I used `/usr/local/bin` but really it is dealers choice (pick something logical from `echo $PATH`). It didn't matter if the copied binary was owned by root or `atuin` user, but I found if you instead attempted to created a symlink with `ln -s` of the binary the permissions with the `atuin` user would still throw a `permission denied` in `journalctl -u atuin-server.service | tail -20`. I am thinking this might be by design with systemd (or the hardening settings in the .service file, but I don't care to figure it out.
  - at this point ran `sudo systemctl start atuin-server`, which it will fail, but it will create the `/etc/atuin.server.toml` file for you now.
  - go edit the `server.toml` file with the appropriate values for postgres and all that and the sucker started up on the next `sudo systemctl restart atuin-server` call. 
  
Basically, the documentation needs to be updated to outline the process here (or something that at least alerts users to the nuances of running your own sync server as a systemd service). 

I can do that if the admins want me to (I don't remember if this is FOSS/licensed for contributors, but I am guessing that it is).

**Comment by sascha-wi** [comment]
Wow what a hassle, I got i working with your instructions. Would be really nice if somebody can edit the documentation.

**Comment by ghostheory** [comment]
Glad to hear you got it running. If I find time this month I might sniff around and see if I can bother to do this.

**Comment by axelson** [comment]
> * create the `/etc/atuin` directory
> * chown/chmod the `/etc/atuin` directory so it was owned by the `atuin` system user I just created (otherwise systemd chokes on bad permissions)
> * `cp` the atuin binary to a path found in `$PATH` so systemd can run it

Thank you! I had to do those same steps as well.

**Comment by markovejnovic** [comment]
Between https://github.com/atuinsh/atuin/pull/1879 and https://github.com/atuinsh/atuin/pull/3176, this should be made easier! Thank you all
